### PR TITLE
Use renewal date from Insurely as suggested start date

### PR DESCRIPTION
**Pull request by robinandeer**
## What?

Use renewal date for the old insurance from Insurely data collection to suggest start date on Offer page.

Add new reusable `useExternalDataCollection` hook. Refactor `InsuranceSelector` to use new hook.

Refactor `StartDate` component and add button to update start date to match renewal date of old insurance.

## Why?

We want to help people switch and BankSignering requires us to send a date for the start of the new insurance.

This only works for Car now but when we can match exposure for other insurance types we can enable this same feature.

There's no way to set a default date in our date picker -- that's why I went with a button the user has to click. This is also to inform the user where the date comes from. I made the button disabled if the start date already matches the renewal date to highlight the same thing.

**Ticket(s): []**

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1185** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/web-onboarding/1185" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/web-onboarding/1185?utm_source=stack-comment).

File: src/client/utils/hooks/useExternalDataCollection.ts
Comment (alebedev) [inline src/client/utils/hooks/useExternalDataCollection.ts:40]: Nit: module name does not prepare me to see other exported things than `useExternalDataCollection` hook.  Can we rename or split?

**Review by alebedev — APPROVED** [review]
Looks good

**Comment by gustaveen** [comment]
I tried to test it in the review app, should work by just adding `CAR_PRICE_MATCHING` and `INSURELY_CAR_CLIENT_ID` env var in the review app right in Heroku? 🤔

File: src/client/utils/hooks/useExternalDataCollection.ts
Comment (robinandeer) [reply src/client/utils/hooks/useExternalDataCollection.ts]: oh good point, those should not be exported

**Review by robinandeer — COMMENTED** [review]
@gustaveen yes that should do it -- I've updated it and deploying again!

File: src/client/pages/Offer/Introduction/Sidebar/StartDate.tsx
Comment (gustaveen) [inline src/client/pages/Offer/Introduction/Sidebar/StartDate.tsx:516]: Shouldn't it be :)
```suggestion
const useGetDateLabel = () => {
```

**Review by gustaveen — APPROVED** [review]
Couldn't test since ICA was unavailable at the moment so hard to validate, but since it's under a feature flag we can do it later 🙌

**Review by guilhermespopolin — APPROVED** [review]

**Comment by robinandeer** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/web-onboarding/1185?utm_source=stack-comment).


Job Reference: npaqwUoQy0fnuqNzCmUF

**Pull request merged**
